lib/Sema/CSApply.cpp

```diff
@@ -6545,7 +6501,7 @@ Expr *ExprRewriter::coerceToType(Expr *expr, Type toType,
       // ```
       //
       // See also: https://bugs.swift.org/browse/SR-6796
-      if (cs.getASTContext().isSwiftVersionAtLeast(3) &&
+      if (cs.getASTContext().isSwiftVersionAtLeast(4) &&
```

Comment:
Isn’t the first part always true? It should suffice to just check if you’re < 5

Reply:
That was intentional. If there are tests passing -swift-version 3 explicitly they should break if they are relying on this behavior. You might have removed all of those with your PR, though.